builder/openstack/step_source_image_info.go

```diff
@@ -117,5 +179,25 @@ func (s *StepSourceImageInfo) Run(ctx context.Context, state multistep.StateBag)
 }
 
 func (s *StepSourceImageInfo) Cleanup(state multistep.StateBag) {
-	// No cleanup required for backout
+	config := state.Get("config").(*Config)
```

Comment:
You can skip the entire clean up if ExternalSourceImageURL is not set.
```suggestion
        if s.ExternalSourceImageURL == "" {
                return 
         }
	config := state.Get("config").(*Config)
```

Reply:
Yes, make sense

Reply:
I guess you would still keep the `if s.ExternalSourceImageURL != "" {}` round the actual deletion since dropping it could cause errors when there's more cleanup added in future versions?

Reply:
In that case, the early check for the ExternalSourceImageURL will be changed as well if necessary in the future. 
I thought about moving up the check because the client creation would be useless if nothing is done, so I think it should be created only if ExternalSourceImageURL is set. 
You can move the client creation inside the `if s.ExternalSourceImageURL != "" ` if you prefer.

Reply:
I think moving the client creation inside the existing if is probably a bit more robust than the guard clause since it doesn't require future editors of that method to understand which part of the code is protected by the guard clause.